### ISSUE TITLE
Scheduler using `clone` on a double-reference

### ISSUE THREAD

**Issue by bgourlie** [bevyengine/bevy#252]
Clippy flagged this error:

```
error: using `clone` on a double-reference; this will copy the reference instead of cloning the inner type
  --> crates\bevy_render\src\render_graph\schedule.rs:78:53
   |
78 |         node_borrows.sort_by_key(|(_node, indices)| indices.clone());
   |                                                     ^^^^^^^^^^^^^^^
   |
   = note: `#[deny(clippy::clone_double_ref)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_double_ref
help: try dereferencing it
   |
78 |         node_borrows.sort_by_key(|(_node, indices)| &(*indices).clone());
   |                                                     ^^^^^^^^^^^^^^^^^^^
help: or try being explicit if you are sure, that you want to clone a reference
   |
78 |         node_borrows.sort_by_key(|(_node, indices)| <&render_graph::schedule::NodeIndices>::clone(indices));
   |                          
```

Seems like a pretty trivial fix once we clarify the intention of the clone.

**Comment by GabLotus** [comment]
Thanks for the issues you're raising, code quality should always remain a top priority!

**Comment by multun** [comment]
I believe this is intended, trying to clone the referenced value doesn't compile at all

**Comment by cart** [comment]
Closed via #206 (although i honestly prefer the old version, clippy is weird sometimes :smile: )
